IPython/core/display.py

```diff
@@ -140,6 +141,88 @@ def _new_id():
     return b2a_hex(os.urandom(16)).decode('ascii')
 
 
+class _RecursiveObject:
+    """
+    Default recursive object that provides a recursion repr if needed.
+    """
+
+    def __repr__(self):
+        return '<recursion ...>'
+
+    def _repr_html_(self):
+        return '&lt;recursion ...&gt;'
+
+
+DataMetadata = namedtuple('DataMetadata', ['data', 'metadata'])
+
+_objs = set()
+
+
+
+def get_repr(obj, include=None, exclude=None, *, on_recursion=_RecursiveObject):
```

Comment:
Naming: should this be something like `get_repr_mimebundle()` or `get_rich_reprs()`? It seems like a good idea to clearly distinguish our rich repr system from Python's text-based reprs.